### PR TITLE
Setup ride preparation, ride and ride completion workflow

### LINKED ISSUE #24: Setup all screens and a complete navigation stack
**Description:**
Implement the basic screens and a meaningful navigation stack.

**Acceptance criteria:**

- [x] Setup all basic screens
- [x] Implement distinct navigation stacks for `Ride Preparation`, `Ride` and *Ride Completion`

### PR DESCRIPTION
**Pull request by nschoen**
This PR sets up the *Ride Preparation*, *Ride* and *Ride Completion* screens and their respective Navigation stacks. Additionally a prototype Workflow/UI has been implemented.
Resolves #24

### CODE REVIEW

File: screens/Ride/ControlScreen.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import {
+  StyleSheet,
+  View,
+  Text,
+  TouchableOpacity,
+} from 'react-native';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+
+const MAX_FAN_LEVEL = 10;
```
Comment (calvinschmidt) [inline screens/Ride/ControlScreen.js:10]: We should outsource all constants in a dedicated file. For now it's ok though.

File: components/MapRoute.js [hunk 2]
```diff
@@ -210,6 +210,7 @@ class MapRoute extends React.Component {
   };
 
   renderConfirmalButton() {
+    const { onArrivalConfirmed } = this.props;
```
Comment (calvinschmidt) [inline components/MapRoute.js:214]: Please add `propTypes` for the `MapRoute` component and add `onArrivalConfirmed`.

File: screens/Ride/ControlScreen.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+class ControlScreen extends React.Component {
+  static navigationOptions = {
+    title: 'Groundcontrol',
+  };
+
+  constructor() {
+    super();
+
+    this.state = {
+      fanLevel: 3,
```
Comment (calvinschmidt) [inline screens/Ride/ControlScreen.js]: Probably also a constant, e.g. `INITIAL_FAN_LEVEL`?

File: screens/Ride/RideScreen.js [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+              {nextStopShift && !nextStopShiftReached && (
+                <Text style={{ color: 'white' }}>Next: End Shift, drive to interchange point</Text>
+              )}
+              {nextStopShiftReached && (
+                <Text style={{ color: 'white' }}>Interchange point reached, end of shift</Text>
+              )}
+            </View>
+          </TouchableOpacity>
+
+          {!nextStopShiftReached && ((autopilotEnabled && (
```
Comment (calvinschmidt) [inline screens/Ride/RideScreen.js]: I had a bit of trouble with these conditions in JSX. Personally, I find these quite unreadable. In general, I would prefer to always refactor conditional JSX into custom render functions that then perform these checks.

**Review by calvinschmidt — COMMENTED** [review]
Looks good, thanks! I just pointed out some minor things.

File: screens/Ride/ControlScreen.js (hunk 3 above)
Comment (nschoen) [reply screens/Ride/ControlScreen.js]: Ok, I will do that for now. But we should think about if this value will be provided by the backend.

**Review by captainsano — APPROVED** [review]

**Pull request merged**